src/LiveComponent/src/DependencyInjection/LiveComponentExtension.php

```diff
@@ -60,15 +61,19 @@ function (ChildDefinition $definition, AsLiveComponent $attribute) {
             ->addTag('twig.component.property_hydrator')
         ;
 
-        $container->register('ux.live_component.doctrine_entity_property_hydrator', DoctrineEntityPropertyHydrator::class)
-            ->setArguments([[new Reference('doctrine')]])
-            ->addTag('twig.component.property_hydrator', ['priority' => -100])
-        ;
+        if (isset($container->getParameter('kernel.bundles')['DoctrineBundle'])) {
+            $container->register('ux.live_component.doctrine_entity_property_hydrator', DoctrineEntityPropertyHydrator::class)
+                ->setArguments([[new Reference('doctrine')]])
+                ->addTag('twig.component.property_hydrator', ['priority' => -100])
+            ;
+        }
 
-        $container->register('ux.live_component.serializer_property_hydrator', NormalizerBridgePropertyHydrator::class)
-            ->setArguments([new Reference('serializer')])
-            ->addTag('twig.component.property_hydrator', ['priority' => -200])
-        ;
+        if (class_exists(Serializer::class)) {
```

Comment:
This will still be an issue if the serializer is installed but not enabled. Is there a way to make this check without a compiler pass?

Reply:
Not that I'm aware of. I think we should just "bite" and move both of these to a compiler pass.